clusterstate/provider/kvm.go

```diff
@@ -0,0 +1,86 @@
+package provider
+
+import (
+	"github.com/giantswarm/e2e-harness/pkg/framework"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type KVMConfig struct {
+	HostFramework *framework.Host
+	Logger        micrologger.Logger
+
+	ClusterID string
+}
+
+type KVM struct {
+	hostFramework *framework.Host
+	logger        micrologger.Logger
+
+	clusterID string
+}
+
+func NewKVM(config KVMConfig) (*KVM, error) {
+	if config.HostFramework == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.HostFramework must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+
+	if config.ClusterID == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.ClusterID must not be empty", config)
+	}
+
+	a := &KVM{
+		hostFramework: config.HostFramework,
+		logger:        config.Logger,
+
+		clusterID: config.ClusterID,
+	}
+
+	return a, nil
+}
+
+func (k *KVM) RebootMaster() error {
+	err := k.deleteMasterPod()
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	return nil
+}
```

Comment:
KVM provider is simplistic due to my lack of knowledge. It just deletes the master pod for both rebooting and replacing the master.

I wasn't sure how to restart the master. Maybe via SSH? Suggestions here would be appreciated. <3